cmd/tsidp/tsidp.go

```diff
@@ -404,6 +433,96 @@ func (s *idpServer) authorize(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	clientID := uq.Get("client_id")
+	if clientID == "" {
+		http.Error(w, "tsidp: must specify client_id", http.StatusBadRequest)
+		return
+	}
+
+	if !s.allowInsecureRegistration {
+		// In strict mode (when insecure registration is NOT allowed), validate client_id exists but defer client_secret validation to token endpoint
```

Comment:
As we're dropped the variables called strict mode, we can do so in the comments as well to remove the need for readers to understand this term. There are lots of comments referring to strict mode, they should refer to allowInsecureRegistration instead